### PR TITLE
test: add remaining tests to v10

### LINKED ISSUE #3229: feat: port all tests from v9 to v10
## Describe the feature you'd like to request

We have a lot of tests that are outdated and using v9. We should have them all in v10-structure.

Good intro if you are curious about contributing to tRPC.

### Instructions

- https://github.com/trpc/trpc/blob/main/CONTRIBUTING.md
- Rewrite all of https://github.com/trpc/trpc/tree/main/packages/tests/server/interop to a nice v10 equivalent (the v10 tests are in the folder above)
- Some have equivalents already
- They can be done one-by-one
- Improvements on the general testing structure is also welcome

#### Example

1. Copy `/packages/tests/server/interop/outputParser.test.ts` -> `/packages/tests/server/outputParser.test.ts` 
1. Replace `legacyRouterToServerAndClient` with `routerToServerAndClientNew`
1. Rewrite the routers using tRPC 10
1. Rewrite the calls using tRPC 10

### Get paid

- We can pay $5-10 per test file rewritten if done well. More for bigger ones (like subscriptions and WebSockets)
- Once you get into the flow, it might not take more than 5 minutes per test file.

### Full tree

```
├── adapters
│   ├── express.test.tsx
│   ├── fastify.test.ts
│   ├── fetch.test.ts
│   └── next.test.ts
├── clientInternals.test.ts
├── dataloader.test.ts
├── errors.test.ts
├── headers.test.tsx
├── index.test.tsx
├── inference.test.ts
├── links.test.ts
├── middleware.test.ts
├── outputParser.test.ts
├── rawFetch.test.tsx
├── react
│   ├── __testHelpers.tsx
│   ├── dehydrate.test.tsx
│   ├── formatError.test.tsx
│   ├── infiniteQuery.test.tsx
│   ├── invalidateQueries.test.tsx
│   ├── prefetchQuery.test.tsx
│   ├── regression
│   │   └── issue-1645-setErrorStatusSSR.test.tsx
│   ├── setInfiniteQueryData.test.tsx
│   ├── setQueryData.test.tsx
│   ├── useMutation.test.tsx
│   ├── useQuery.test.tsx
│   ├── useSubscription.test.tsx
│   └── withTRPC.test.tsx
├── regression
│   ├── issue-949-inferProcedureInput.test.ts
│   └── issue-990-initialData.test.tsx
├── responseMeta.test.ts
├── router.test.ts
├── routerMeta.test.ts
├── transformer.test.ts
├── validators.test.ts
```

### PR DESCRIPTION
**Pull request by Nsttt**
Closes #3229

## 🎯 Changes

Add remaining tests to v10

## ✅ Checklist

- [x] I have followed the steps listed in the [Contributing guide](https://github.com/trpc/trpc/blob/main/CONTRIBUTING.md).
- [x] If necessary, I have added documentation related to the changes made.
- [x] I have added or updated the tests related to the changes made.

### CODE REVIEW

**Comment by Nsttt** [comment]
There's 2 cases right now that for some reason doesn't behave properly in `index.test.ts`. I've marked them for now, will come back to them later.

File: packages/tests/server/errors.test.ts [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+//     toString: t.procedure.query(() => 'toStringValue'),
+//     hasOwnProperty: t.procedure.query(() => 'hasOwnPropertyValue'),
+//   });
+
+//   const { close, proxy } = routerToServerAndClientNew(router);
+
+//   expect(await proxy.toString.query()).toBe('toStringValue');
+//   expect(await proxy.hasOwnProperty.query()).toBe('hasOwnPropertyValue');
+//   close();
+// });
```
Comment (Nsttt) [inline packages/tests/server/errors.test.ts:330]: These are the two test that keep failing due to regresion apparently.

File: packages/tests/server/index.test.ts [hunk 2]
```diff
@@ -0,0 +1,597 @@
...
+    // !  proxy.q.query('not-nullish' as any),
+    // ! ).rejects.toMatchInlineSnapshot(`[TRPCClientError: No input expected]`);
+
+    await proxy.m.mutate();
+    await proxy.m.mutate(undefined);
+    await proxy.m.mutate(null as any); // treat null as undefined
+    // ! This tests doesn't work
+    // ! await expect(
+    // !  proxy.m.mutate('not-nullish' as any),
+    // ! ).rejects.toMatchInlineSnapshot(`[TRPCClientError: No input expected]`);
```
Comment (Nsttt) [inline packages/tests/server/index.test.ts:169]: This are the other 2 that fail as well.

**Comment by KATT** [comment]
Just converting to draft so we don't accidentally merge this before looking at the failing tests

File: packages/client/src/createTRPCClientProxy.ts [hunk 3]
```diff
@@ -99,7 +99,7 @@ export function createTRPCClientProxy<TRouter extends AnyRouter>(
   client: TRPCClient<TRouter>,
 ) {
   return createFlatProxy<CreateTRPCProxyClient<TRouter>>((key) => {
-    if (key in client) {
+    if (client.hasOwnProperty(key)) {
```
Comment (KATT) [inline packages/client/src/createTRPCClientProxy.ts:102]: 🐛

**Review by KATT — APPROVED** [review]

File: packages/client/src/createTRPCClientProxy.ts (hunk 3 above)
Comment (Nsttt) [reply packages/client/src/createTRPCClientProxy.ts:102]: 😯

**Review by KATT — DISMISSED** [review]

**Review by KATT — APPROVED** [review]

**Pull request merged**